include/drivers/rtc.h

```diff
@@ -0,0 +1,262 @@
+/**
+ * @file
+ * @brief RTC public API header file.
+ */
+/*
+ * Copyright (c) 2021 LACROIX Group
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_RTC_H_
+#define ZEPHYR_INCLUDE_DRIVERS_RTC_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <time.h>
+#include <device.h>
+// #include <dt-bindings/rtc/rtc.h>
+
+/**
+ * @brief RTC driver APIs
+ * @defgroup rtc_interface RTC driver APIs
+ * @ingroup io_interfaces
+ * @{
+ */
+
+/** @brief Wake up timers enum. */
+enum rtc_wakeup_id {
+	RTC_WUT0,       /**< Wake Up timer 0. */
+	/* Add new item before RTC_WUT_NUM. */
```

Comment:
This doesn't scale very well - it should not be required to add enumerations to a global header just to index a wakeup id.

Reply:
Out of curiosity, have you ever seen a RTC with more than 1 wakeup timer? Seems unnecessary since only the wakeup timer with the shortest period has any influence on wakeup. Since the RTC must be ultra low power, adding another one would just increase power consumption without adding meaningful functionality.

I would remove the ID, unless anyone knows of an RTC with more than one wakeup timer :)

Reply:
http://ww1.microchip.com/downloads/en/devicedoc/20005010f.pdf

> Dual Programmable Alarms

Reply:
The alarms are different from wakeup timers, most RTCs have more than one alarm, but only one wakeup timer if any. The Wakeup timer just counts up to a match value, and then invokes an interrupt, which can be mapped to wake up the device. The point of the timer is to be very low power, and as such, only has this limited capability.
Alarms on the other hand use the date and time to invoke alarms, and require much more power.

The device linked here has to alarms, no wakeup timer http://ww1.microchip.com/downloads/en/devicedoc/20005010f.pdf

Reply:
Alternately, just use DT chosen, and then the ID is implicit.